.github/workflows/beam_LoadTests_Go_GBK_Dataflow_Batch.yml

```diff
@@ -0,0 +1,153 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Load Tests GBK Dataflow Batch Go
+
+on:
+  issue_comment:
+    types: [created]
+  schedule:
+    - cron: '40 23 * * *'
+  workflow_dispatch:
+
+#Setting explicit permissions for the action to avoid the default permissions which are `write-all` in case of pull_request_target event
+permissions:
+  actions: write
+  pull-requests: read
+  checks: read
+  contents: read
+  deployments: read
+  id-token: none
+  issues: read
+  discussions: read
+  packages: read
+  pages: read
+  repository-projects: read
+  security-events: read
+  statuses: read
+
+# This allows a subsequently queued workflow run to interrupt previous runs
+concurrency:
+  group: '${{ github.workflow }} @ ${{ github.event.issue.number || github.sha || github.head_ref || github.ref }}-${{ github.event.schedule || github.event.comment.body || github.event.sender.login }}'
+  cancel-in-progress: true
+
+env:
+  GRADLE_ENTERPRISE_ACCESS_KEY: ${{ secrets.GE_ACCESS_TOKEN }}
+  GRADLE_ENTERPRISE_CACHE_USERNAME: ${{ secrets.GE_CACHE_USERNAME }}
+  GRADLE_ENTERPRISE_CACHE_PASSWORD: ${{ secrets.GE_CACHE_PASSWORD }}
+
+jobs:
+  beam_LoadTests_Go_GBK_Dataflow_Batch:
+    if: |
+      github.event_name == 'workflow_dispatch' ||
+      github.event_name == 'schedule' ||
+      github.event.comment.body == 'Run Load Tests Go GBK Dataflow Batch'
+    runs-on: [self-hosted, ubuntu-20.04, main]
+    timeout-minutes: 720
+    name: ${{ matrix.job_name }} (${{ matrix.job_phrase }})
+    strategy:
+      matrix:
+        job_name: ["beam_LoadTests_Go_GBK_Dataflow_Batch"]
+        job_phrase: ["Run Load Tests Go GBK Dataflow Batch"]
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup repository
+        uses: ./.github/actions/setup-action
+        with:
+          comment_phrase: ${{ matrix.job_phrase }}
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          github_job: ${{ matrix.job_name }} (${{ matrix.job_phrase }})
+      - name: Prepare configs
+        id: set_configs
+        shell: bash
+        run: |
+          CURDATE=$(date '+%m%d%H%M%S' --utc)
+          CURCONFIG=$(grep -v "^#.*" ./.github/workflows/load-tests-job-configs/config_GBK_Go_Batch_10b.txt | tr '\n' ' ')
```

Comment:
I'm not sure about the correct answer, but could we implement a matrix of file names here and execute them all in parallel? Is this suitable for load testing? 
For example, a matrix could be: file_names: [config_GBK_Go_Batch_10b.txt, config_GBK_Go_Batch_100b.txt, ..., etc]. The other steps seem similar.
Thank you.

Reply:
I think no, because: 
- there is no need to run 2MB test if 10KB test is failing
- sometimes in load tests we have steps that should be done before one bunch of tests (e.g. start flink) and some steps before another bunch, and some steps that should be done only after all tests (e.g. teardown flink)

Reply:
I agree with not doing it via matrix, but could we simplify this with a list/loop? If its ugly in bash, we could also consider https://github.com/actions/github-script

Reply:
Done